src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -155,8 +189,27 @@ public virtual string RouteUrl(UrlRouteContext routeContext)
             }
 
             var valuesDictionary = routeContext.Values as RouteValueDictionary ?? GetValuesDictionary(routeContext.Values);
-            var virtualPathData = GetVirtualPathData(routeContext.RouteName, valuesDictionary);
-            return GenerateUrl(routeContext.Protocol, routeContext.Host, virtualPathData, routeContext.Fragment);
+
+            if (_linkGenerator == null)
+            {
+                var virtualPathData = GetVirtualPathData(routeContext.RouteName, valuesDictionary);
+                return GenerateUrl(routeContext.Protocol, routeContext.Host, virtualPathData, routeContext.Fragment);
+            }
+            else
+            {
+                var canGetLink = _linkGenerator.TryGetLink(
+                    new LinkGeneratorContext()
+                    {
+                        Address = new Address(routeContext.RouteName),
+                        SuppliedValues = valuesDictionary,
+                        AmbientValues = AmbientValues
+                    },
+                    out var link);
+
+                if (!canGetLink) { return null; }
```

Comment:
eww

Reply:
Eww for?

Returning null is the existing behavior right?

Reply:
I believe he meant the braces should be put in their own lines :-)